tools/clang/lib/SPIRV/EmitSPIRVAction.cpp

```diff
@@ -58,14 +93,23 @@ class SPIRVEmitter : public ASTConsumer {
     const uint32_t funcType = translateFunctionType(decl);
     const uint32_t retType = translateType(decl->getReturnType());
 
-    theBuilder.beginFunction(funcType, retType);
+    const uint32_t funcId = theBuilder.beginFunction(funcType, retType);
     // TODO: handle function parameters
     // TODO: handle function body
     const uint32_t entryLabel = theBuilder.bbCreate();
     theBuilder.bbReturn(entryLabel);
     theBuilder.endFunction();
-  }
 
+    // Add an entry point to the module if necessary
+    const std::string hlslEntryFn =
+        theCompilerInstance.getCodeGenOpts().HLSLEntryFunction;
+    if (hlslEntryFn == decl->getNameInfo().getAsString()) {
+      const spv::ExecutionModel em = getSpirvShaderKindFromHlslProfile(
+          theCompilerInstance.getCodeGenOpts().HLSLProfile.c_str());
+      spirv::EntryPoint ep(em, funcId, hlslEntryFn, /* Interfaces (TODO) */ {});
```

Comment:
There is an existing `addEntryPoint()` method on `SPIRVModule` that constructs `EntryPoint` internally. We should use that to avoid pulling in too many definitions from `Structure.h`.